gtsam/nonlinear/factorTesting.h

```diff
@@ -46,24 +45,24 @@ JacobianFactor linearizeNumerically(const NoiseModelFactor& factor,
 
   // Loop over all variables
   const double one_over_2delta = 1.0 / (2.0 * delta);
-  for(Key key: factor) {
+  for (Key key : factor) {
     // Compute central differences using the values struct.
     VectorValues dX = values.zeroVectors();
     const size_t cols = dX.dim(key);
     Matrix J = Matrix::Zero(rows, cols);
     for (size_t col = 0; col < cols; ++col) {
       Eigen::VectorXd dx = Eigen::VectorXd::Zero(cols);
-      dx[col] = delta;
+      dx(col) = delta;
       dX[key] = dx;
       Values eval_values = values.retract(dX);
       const Eigen::VectorXd left = factor.whitenedError(eval_values);
-      dx[col] = -delta;
+      dx(col) = -delta;
```

Comment:
what's the preference here between square bracket and paren?

Reply:
Parens allow for safer access in Eigen matrices and is the preferred syntax.

Checkout the section `Coefficient accessors` at [this link](https://eigen.tuxfamily.org/dox/group__TutorialMatrixClass.html).

Reply:
thanks!